gramps/gen/filters/test/filter_optimizer_test.py

```diff
@@ -183,9 +182,12 @@ def test_everyone_with_id_list_tupleind_3(self):
         filter = self.filters["Everyone"]
         default_person = self.db.get_default_person()
         id_list = [(default_person, default_person.handle)]
-        with pytest.raises(IndexError) as excinfo:
```

Comment:
You can use [self.assertRaises](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises) here.